src/nodes.coffee

```diff
@@ -2007,7 +2007,11 @@ exports.ExecutableClassBody = class ExecutableClassBody extends Base
         # The class scope is not available yet, so return the assignment to update later
         assign = @externalCtor = new Assign new Value, value
       else if not assign.variable.this
-        name      = new (if base.shouldCache() then Index else Access) base
+        name =
```

Comment:
This is the fix for the invalid JS for non-static computed class properties (extracted from #5205)